src/main/java/seedu/address/model/scheduler/time/HourBlock.java

```diff
@@ -73,6 +75,7 @@ public boolean isFree() {
      * @param commitment
      */
     public void setCommitment(Commitment commitment) {
+        requireNonNull(commitment);
```

Comment:
Good use of `requireNonNull` here

Reply:
Agreed. Seems like we’ve been doing some good defensive programming since the start.